src/Raven.Server/Rachis/Follower.cs

```diff
@@ -602,10 +605,28 @@ private void ReadAndCommitSnapshot(TransactionOperationContext context, InstallS
             }
         }
 
-        private unsafe bool InstallSnapshot(TransactionOperationContext context, CancellationToken token)
+        private bool InstallSnapshot(TransactionOperationContext context, CancellationToken token)
         {
             var txw = context.Transaction.InnerTransaction;
-            var reader = _connection.CreateReader();
+
+            var fileName = $"snapshot.{Guid.NewGuid():N}";
+            var filePath = context.Environment.Options.DataPager.Options.TempPath.Combine(fileName);
+            using (var temp = new StreamsTempFile(filePath.FullPath, context.Environment))
+            using(var stream = temp.StartNewStream())
+            {
+                var remoteReader = _connection.CreateReaderToStream(stream);
+
+                if (ReadSnapshotFromFile(remoteReader, context, txw,false, token) == false)
+                    return false;
+
+                stream.Seek(0, SeekOrigin.Begin);
+                var fileReader = new RemoteConnection.StreamReader(stream);
+                return ReadSnapshotFromFile(fileReader, context, txw, true, token);
+            }
+        }
+        
+        private unsafe bool ReadSnapshotFromFile(RemoteConnection.Reader reader, TransactionOperationContext context, Transaction txw, bool doWork, CancellationToken token)
```

Comment:
`doWork` name need to explain what it does better

Reply:
`dryRun` ?

Reply:
I like this